### ISSUE TITLE
INI Question (Enhancement?): Having ; in value?

### ISSUE THREAD

**Issue by zakaluka** [fsprojects/FSharp.Configuration#142]
Hello,

Is there a way to place a `;` in a value such that the INI parser won't interpret it as a comment?

I have tried the following without success:

```ini
Migrations=Data Source=rmlaptop-140\sqlexpress;Initial Catalog=ZenCase
Migrations="Data Source=rmlaptop-140\sqlexpress;Initial Catalog=ZenCase"
Migrations='Data Source=rmlaptop-140\sqlexpress;Initial Catalog=ZenCase'
Migrations=Data Source=rmlaptop-140\sqlexpress\;Initial Catalog=ZenCase
```

There's some more variations that I also tried without success.

Finally, read the source code for the INI provider and saw that the Regex specifically prohibits `;` in the value `\s*(\S+)\s*=\s*([^;]*)`.

Would it be possible to either support quoted characters (i.e. `\;`) or to not treat mid-value semicolons as comments?

**Comment by zakaluka** [comment]
After beating my head for a couple of hours against the wall, 2 options stand out:
1. Only treat `;` as a comment if it is preceded by some sort of whitespace
1. Don't allow comments at the end of a line (i.e. only support comments when the `;` is the first character in the line)

**Comment by zakaluka** [comment]
If there is a preference for a solution, I'm happy to submit a PR.

**Comment by baronfel** [comment]
https://en.wikipedia.org/wiki/INI_file#Comments Suggests to me that the comment-detection regex should look for `;` only at the start of a line, and then we could remove the `;` check entirely from the `String` matcher.

**Comment by zakaluka** [comment]
That's pretty much where I got my options from.  I'll submit a PR with that in mind (it's a pretty minor change to the regex) and write some tests to ensure it works.

**Comment by baronfel** [comment]
Thank you!

**Comment by zakaluka** [comment]
Please note that given the following Wikipedia statement and the fact that the INI type provider currently ignores leading whitespace from a value, I have also removed trailing whitespace from the value to be consistent (and added the corresponding test).

> Some implementations also ignore leading and trailing whitespace around the property value; others consider all characters following the equals sign (including whitespace) to be part of the value.

**Comment by baronfel** [comment]
Published as version 1.5.0
